src/components/content/deployedServices/myServices/MyServiceHistory.tsx

```diff
@@ -92,22 +136,45 @@ export const MyServiceHistory = ({
             dataIndex: 'startedTime',
             align: 'center',
             width: 150,
+            defaultSortOrder: 'descend',
+            sorter: (serviceOrderDetailsA, serviceOrderDetailsB): number => {
+                if (serviceOrderDetailsA.startedTime && serviceOrderDetailsB.startedTime) {
+                    return (
+                        new Date(serviceOrderDetailsA.startedTime).getTime() -
+                        new Date(serviceOrderDetailsB.startedTime).getTime()
+                    );
+                }
+                return 0;
+            },
         },
         {
             title: 'CompletedTime',
             dataIndex: 'completedTime',
             align: 'center',
             width: 150,
+            sorter: (serviceOrderDetailsA, serviceOrderDetailsB): number => {
+                if (serviceOrderDetailsA.completedTime && serviceOrderDetailsB.completedTime) {
+                    return (
+                        new Date(serviceOrderDetailsA.completedTime).getTime() -
+                        new Date(serviceOrderDetailsB.completedTime).getTime()
+                    );
+                }
+                return 0;
+            },
         },
         {
-            title: 'Status',
+            title: 'TaskStatus',
             dataIndex: 'taskStatus',
             align: 'center',
             width: 50,
+            filters: taskStatusFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: React.Key | boolean, record) => record.taskStatus.startsWith(value.toString()),
             render: (value) => {
                 if (value === taskStatus.FAILED) {
```

Comment:
we need some better handling for other two task states
CREATED("created"), - can be in grey color with a corresponding icon
IN_PROGRESS("in-progress"), - orange and loading icon

Reply:
fixes like this:
![image](https://github.com/user-attachments/assets/472028b0-c3f0-49f6-9e66-1c6d14f17244)

![image](https://github.com/user-attachments/assets/c9e35192-d694-4837-8151-2ed6b1feec2b)